src/plugins/graph/components/editable-graph-value.tsx

```diff
@@ -0,0 +1,128 @@
+import { observer } from 'mobx-react';
+import React, { useState, useRef, useEffect, CSSProperties } from 'react';
+import { AxisBounds, AxisPlace } from '../imports/components/axis/axis-types';
+import { useAxisLayoutContext } from '../imports/components/axis/models/axis-layout-context';
+
+import "./editable-graph-value.scss";
+
+interface IEditableValueProps {
+  value: number;
+  minOrMax: "min" | "max";
+  axis: AxisPlace;
+  onValueChange: (newValue: number) => void;
+  readOnly?: boolean;
+}
+
+export const EditableGraphValue: React.FC<IEditableValueProps> = observer(function EditableGraphValue(props) {
+  const { value, minOrMax, axis, onValueChange, readOnly } = props;
+  const [isEditing, setIsEditing] = useState(false);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const layout = useAxisLayoutContext();
+  const axisBounds = layout.getComputedBounds(axis) as AxisBounds;
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.select();
+    }
+  }, [isEditing]);
+
+  //**************  Calculate border box size and offset positions ******************
+
+  const calculateBorderBoxStyles = () => {
+    const numOfCharacters = value.toString().length;
+    const widthPerCharacter = 8;
+    let boxWidth: number;
+    // Calculate boxWidth based on the number of characters
+    if (numOfCharacters === 1) {
+      boxWidth = 15;
+    } else if (numOfCharacters === 2) {
+      boxWidth = 20;
+    } else {
+      boxWidth = numOfCharacters * widthPerCharacter;
+    }
+
+    const style: CSSProperties = {
+      width: `${boxWidth}px`,
+      justifyContent: axis === 'bottom' ? 'center' : 'flex-end',
```

Comment:
I just added this - such that the vertical Y axis <EditableGraphValue> boxes are right aligned, and the bottom ones are centered. This makes for a cleaner alignment when I drag and add big numbers - the last digit is always aligned with the other tick values.